micro-rdk/src/common/ota.rs

```diff
@@ -524,9 +511,11 @@ impl<S: OtaMetadataStorage> OtaService<S> {
         // Test experimental ffi accesses here to be recoverable without flashing
         #[cfg(feature = "esp32")]
         {
-            log::info!("rebooting to load firmware from `{:#x}`", self.address);
+            log::info!(
+                "next reboot will load firmware from `{:#x}`",
+                self.address
+            );
             // TODO(RSDK-9464): flush logs to app.viam before restarting
-            esp_idf_svc::hal::reset::restart();
```

Comment:
this shifts the responsibility of rebooting to whoever is calling `ota::update()`